lib/service/user_interactivity_service.dart

```diff
@@ -0,0 +1,81 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:autonomy_flutter/model/dailies.dart';
+import 'package:autonomy_flutter/service/configuration_service.dart';
+import 'package:autonomy_flutter/service/metric_client_service.dart';
+import 'package:autonomy_flutter/util/metric_helper.dart';
+import 'package:autonomy_flutter/util/ui_helper.dart';
+import 'package:easy_localization/easy_localization.dart';
+
+abstract class UserInteractivityService {
+  Future<void> likeDailyWork(DailyToken dailyToken);
+}
+
+class UserInteractivityServiceImpl implements UserInteractivityService {
+  final ConfigurationService _configurationService;
+  final MetricClientService _metricClientService;
+
+  UserInteractivityServiceImpl(
+      this._configurationService, this._metricClientService);
+
+  @override
+  Future<void> likeDailyWork(DailyToken dailyToken) async {
+    final data = {
+      MetricParameter.tokenId: dailyToken.tokenID,
+      MetricParameter.localTime: DateTime.now().toIso8601String(),
+    };
+    unawaited(
+        _metricClientService.addEvent(MetricEventName.dailyLiked, data: data));
+
+    final isNotificationEnabled = _configurationService.isNotificationEnabled();
+    if (!isNotificationEnabled) {
+      final likedCount = _configurationService.getDailyLikedCount();
+      if (likedCount >= 3) {
+        await _showEnableNotificationDialog();
+      } else {
+        await _configurationService.setDailyLikedCount(likedCount + 1);
+      }
+    }
+  }
+
+  Future<dynamic> _showEnableNotificationDialog() async {
+    final type = EnableNotificationPromptType.getRandomType();
+    await UIHelper.showNotificationPrompt(type);
+    await _configurationService.setDailyLikedCount(0);
```

Comment:
Do we show notification after EACH 3 times?

Reply:
that's what i intend